shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -53,21 +70,20 @@ export function createOfflineListener(parentEvtNamespace?: string | string[]): I
     function _enableEvents(target: any): boolean {
         let enabled = false;
         if (target) {
-            enabled = eventOn(target, "online", _setOnline, _evtNamespace);
+            enabled = eventOn(target, "online", _setOnlineState(true), _evtNamespace);
             if (enabled) {
-                eventOn(target, "offline", _setOffline, _evtNamespace);
+                eventOn(target, "offline", _setOnlineState(false), _evtNamespace);
             }
         }
 
         return enabled;
     }
 
-    function _setOnline() {
-        _onlineStatus = true;
-    }
-
-    function _setOffline() {
-        _onlineStatus = false;
+    function _setOnlineState(isOnline: boolean){
+        _onlineStatus = isOnline;
+        listenerList.forEach((callback: OfflineCallback) => {
```

Comment:
This "listenerList" should only be processed "IF" the status changes, not if the state is still considered to be the same.

Reply:
And as `listenerList` should be an array this should be using `arrForEach()`